File: frontend/src/types/tableRequest.ts
export interface TableRequestCreate {
  customer_name: string;
  customer_mobile: string;
}

export interface TableRequestResponse {
  request_id: number;
  table_id: number;
  customer_name: string;
  customer_mobile: string;
  waiter_id: number | null;
  status: string;
  requested_at: string;
  accepted_at: string | null;
}

File: Backend/app/schemas/table_request.py
from datetime import datetime
from pydantic import BaseModel


class TableRequestCreate(BaseModel):

    customer_name: str

    customer_mobile: str


class TableRequestResponse(BaseModel):

    request_id: int

    table_id: int

    customer_name: str

    customer_mobile: str

    waiter_id: int | None

    status: str

    requested_at: datetime

    accepted_at: datetime | None

    class Config:
        from_attributes = True